draft-birkholz-scitt-architecture.md

```diff
@@ -552,64 +575,126 @@ Editor's Note: This may be moved to appendix.
 POST <Base URL>/entries
 ~~~
 
+Headers:
+
+- `Content-Type: application/cose`
+
 Body: SCITT COSE_Sign1 message
 
 #### Response
 
 One of the following:
 
-- HTTP Status `201` - Registration was tentatively successful pending service consensus.
-- HTTP Status `400` - Registration was unsuccessful.
-  - Error code `AwaitingDIDResolutionTryLater`
+- Status 202 - Registration is successful.
```

Comment:
why not 200 here?  Isn't "registered" a final state?

Reply:
The idea is that every submission is uniformly treated as a long-running operation from an API standpoint, even if the operation finishes immediately. This makes clients a little easier to write. Returning the operation object makes really only sense with 202. For 200/201 I would expect either a redirect to the created object or the response being the actual created object. Doing that, clients would have to support both cases.

Reply:
I am not sure I understand the simplicity argument, but regardless that doesn't really match HTTP semantics (see [§15.3.3 of STD97](https://www.rfc-editor.org/rfc/rfc9110.html#name-202-accepted)):

_"[...] the processing has not been completed; in fact, processing may not have started yet."_

Reply:
With simplicity, I mean this: https://github.com/scitt-community/scitt-api-emulator/blob/4a56fad2171a4a0fd32075a5378d21b3aa66aab4/scitt_emulator/client.py#L78-L92. If you have two possible answer types (one creating and returning an operation, the other not) then the code to handle that gets more complicated. I found some reference for this here: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#132-stepwise-long-running-operations "For an API that's defined as a Stepwise Long Running Operation the service MUST go through the Stepwise Long Running Operation flow even if the operation can be completed immediately."
If we have two separate responses, 202 for long-running, and 201 for immediate, which would you return in the 201 case?

Reply:
my preference would be:

### Request

```http
POST /entries HTTP/1.1
Host: scitt.example
Accept: ...
Content-Type: application/cose

[ COSE Sign1 claims ]
```

### Response

#### Immediate

```http
HTTP/1.1 201 Created
Location: /entries/{entry-id}
Content-Type: application/json

{ "entry-representation": "..." }
```

#### Pending

```http
HTTP/1.1 202 Accepted
Location: {operation-monitor-url}
Content-Type: application/json

{ "operation-monitor-resource-representation": "..." }
```

Reply:
Alright, let's do that. I prototyped it in the emulator and didn't see any issue. There is a little more code for clients but it is minimal and we expect this to be hidden in libraries anyway.